configs/ruby/GPU_VIPER.py

```diff
@@ -273,6 +273,8 @@ class TCC(RubyCache):
 
     def create(self, options):
         self.assoc = options.tcc_assoc
+        self.atomicLatency = options.atomic_alu_latency
+        self.atomicALUs = options.tcc_num_atomic_alus
```

Comment:
Are we guaranteed that if these options are provided that each TCC `resourceStalls` will be true?

I.e., would it be possible for the user to specify `--tcc_num_atomic_alus=8` on the command line and then it be ignored by the simulation?

Reply:
I am not aware of this being possible, but I'm also not sure what proof you would want either.

Reply:
The code should probably look something like

```
if <user specifies `--tcc_num_atomic_alus`>:
    for cache in <tcc_sim_object_list>:
        cache.resource_stalls = True
```

Without this, I think you'll find many people misusing this code (i.e., thinking that their command line parameter changes something when it doesn't).

Reply:
Personally I agree with @DanKouch here, but since @powerjg is concerned about this and is away right now, we'll probably have to wait for him to return to comment on this.

Reply:
> It looks like this will happen if `--no-tcc-resource-stalls` is supplied. Since tcc-num-atomic-alus defaults to 64, doing what the above psuedocode specifies would make it so that the `--no-tcc-resource-stalls` is ignored unless `--tcc-num-atomic-alus=0` is also specified, which seems like it might be more confusing. I could have it override if tcc-num-atomic-alus isn't 64, but specifying `--no-tcc-resource-stalls` seems rather explicit.

@powerjg Are you ok with leaving it as-is in light of this?

Reply:
If the default is that you will have resource stalls, then this is OK. But if the default is that resources stalls are off, then this is very confusing for users other than the person who wrote the code.

Reply:
Yes, the default is to have resource stalls.